### PR TITLE
🐛 Source Tiktok: `include_deleted` argument does not affect ads, ad_groups, and campaigns streams

### PR DESCRIPTION
**Pull request by abedrytski**
## What
Linked issue https://github.com/airbytehq/airbyte/issues/32095

The include_deleted parameter is currently described as allowing users to "include deleted data in reports" for the TikTok marketing source. However, when this parameter is set, it appears to have no impact on the ads, ad_groups, and campaigns data streams. The secondary_status column within these streams does not reflect deleted entities with the status "AD_STATUS_DELETE", leading to a lack of visibility on whether an entity has been deleted.

Impact:

Without the secondary_status being updated correctly, users are unable to know the deletion status of ads, ad groups, and campaigns, which can lead to discrepancies in data reporting and analysis.

## How
Make `include_deleted` parameter also apply to ads, ad groups and campaigns streams.

## Recommended reading order
1. `airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py`
2. `airbyte-integrations/connectors/source-tiktok-marketing/unit_tests/streams_test.py`

## 🚨 User Impact 🚨
No breaking changes, but users may find more data ingested for their ads, ad groups and campaigns streams in their destination.

This PR has been tested in production in our company for more than 3 weeks.

### CODE REVIEW

**Review by marcosmarxm — COMMENTED** [review]
Thanks for the contribution @abedrytski, a quick look it makes sense. During the week I'll run tests to validate the change and ask for the connector team to make the final review. The codebase is freeze this week because of holiday and it will be able to merge only next week.

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 1]
```diff
@@ -372,6 +373,9 @@ def request_params(
             params["fields"] = self.convert_array_param(self.fields)
         if stream_slice:
             params.update(stream_slice)
+        if self.include_deleted and self.name in ('ads', 'ad_groups', 'campaigns'):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:376]: I'm not sure that is good idea to filtrate by child names in parent class. Better create some class/mixin with additional functional and add it to child classes.

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py (hunk 1 above)
Comment (abedrytski) [reply airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:376]: Hi @lazebnyi, what do you think about adding an additional field to the class FullRefreshTiktokStream named include_deleted_params with a default value of {}? This value would be overridden in the corresponding classes, so the code above would look like this:

```
        if self.include_deleted:
            params.update(self.include_deleted_params) # or `include_deleted_request_params`
```

And the corresponding class would look like:

```
class Campaigns(IncrementalTiktokStream):
    include_deleted_params = {"filtering": '{"secondary_status": "CAMPAIGN_STATUS_ALL"}'}
    primary_key = "campaign_id"

    def path(self, *args, **kwargs) -> str:
        return "campaign/get/"
        ```

Would that be acceptable? I find mixins a bit overkill, especially considering that they might lead to method overriding and increase code duplication.

That said, this is only a suggestion. If you prefer mixins, I can certainly update the code to use them.

**Comment by abedrytski** [comment]
Hi @lazebnyi, I've updated the PR. Although it doesn't use mixins, I think it solves the problem you mentioned in the feedback. Would that be okay, or would you prefer a solution that uses mixins?

File: airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py [hunk 2]
```diff
@@ -284,6 +284,7 @@ def path(self, *args, **kwargs) -> str:
 class FullRefreshTiktokStream(TiktokStream, ABC):
```
Comment (Shahbaz7000) [inline airbyte-integrations/connectors/source-tiktok-marketing/source_tiktok_marketing/streams.py:284]: @king_0f.7?_t=8m0UXPzQPNO&_r=1

**Comment by abedrytski** [comment]
@holger-vinted It's not up to me to approve this pr.
@lazebnyi Have I resolved the issue you've metionned above? (merge conflicts are only related to the documentation of the release notes)

**Comment by darynaishchenko** [comment]
@abedrytski Hi, `include deleted` was added as part of migration to low code in this pr https://github.com/airbytehq/airbyte/pull/38316. Thanks for contribution, I'll close your pr in favor of migration.

**Comment by ChloeBellm** [comment]
Hi, I'm running the v4.0.0 TikTok source connector and despite toggling on include_deleted I still don't seem to get deleted ads in the `ads_reports_insights` stream. Is this expected to work now and am I doing something wrong?

**Comment by darynaishchenko** [comment]
@ChloeBellm Hi, we don't have `ads_reports_insights` stream in tiktok marketing, only `ads_reports_daily`, `ads_reports_hourly` and `ads_reports_lifetime`. What stream you are referring to? All report streams should support include deleted if `include_deleted` is set to true. You can contact support to create an issue and we can make a futher investigation of your case.

**Pull request closed without merging**